src/ucs/sys/sys.c

```diff
@@ -481,40 +482,42 @@ void ucs_get_mem_info(void *address, size_t size, ucs_mem_info_cb_t cb,
         return;
     }
 
+    /* coverity[tainted_data_argument] */
     while (fgets(buf, sizeof(buf), file) != NULL) {
         n = sscanf(buf, "%lx-%lx", &start, &end);
         if (n != 2) {
             continue;
         }
 
-        if (address != NULL) {
-            if (start > (uintptr_t)address + size) {
-                /* the scanned range is after memory range of interest - stop */
-                break;
-            }
-            if (end <= (uintptr_t)address) {
-                /* the scanned range is still before the memory range of interest */
-                continue;
-            }
+        if (start > (uintptr_t)address + size) {
+            /* the scanned range is after memory range of interest - stop */
+            break;
+        }
+        if (end <= (uintptr_t)address) {
+            /* the scanned range is still before the memory range of interest */
+            continue;
         }
 
+        memset(&info, 0, sizeof(info));
         info.start = start;
         info.end   = end;
 
         while (fgets(buf, sizeof(buf), file) != NULL) {
             n = sscanf(buf, "KernelPageSize: %lu kB", &page_size_kb);
             if (n == 1) {
                 info.page_size = page_size_kb * UCS_KBYTE;
+                continue;
             }
 
             n = 9;
             if (memcmp(buf, "VmFlags: ", n) == 0) {
-                while (buf[n] != '\n' && n < sizeof(buf)) {
-                    if (memcmp(buf + n, "dc", 2) == 0) {
-                        info.flags |= UCS_MEM_FLAG_DONTCOPY;
+                p = buf + n;
+                while ((p = strtok_r(p, " \n", &save))) {
+                    if (strcmp(p, "dc") == 0) {
+                        info.flags |= UCS_SYS_VMA_FLAG_DONTCOPY;
                     }
 
-                    n += 3;
+                    p = NULL;
```

Comment:
why needed?

Reply:
char *strtok_r(char *str, const char *delim, char **saveptr);
In each subsequent call that should parse the same string, str must be NULL.

Reply:
right